src/encoded/static/components/static-pages/placeholders/MdSortableTable.js

```diff
@@ -0,0 +1,643 @@
+            const key = this.props.columns[i].key;
+            switch (sorting) {
+                case "desc":
+                    if (column.descSortFunction &&
+                        typeof (column.descSortFunction) == "function") {
+                        sortedData = column.descSortFunction(sortedData, key);
+                    } else {
+                        sortedData = this.descSortData(sortedData, key);
+                    }
+                    break;
+                case "asc":
+                    if (column.ascSortFunction &&
+                        typeof (column.ascSortFunction) == "function") {
+                        sortedData = column.ascSortFunction(sortedData, key);
+                    } else {
+                        sortedData = this.ascSortData(sortedData, key);
+                    }
+                    break;
+            }
+        }
+        return sortedData;
+    }
+
+    ascSortData(data, key) {
+        return this.sortDataByKey(data, key, ((a, b) => {
+            if (Utils.parseFloatable(a) && Utils.parseFloatable(b)) {
+                a = Utils.parseIfFloat(a);
+                b = Utils.parseIfFloat(b);
+            }
+            if (a >= b) {
+                return 1;
+            } else if (a < b) {
+                return -1;
+            }
+        }).bind(this));
+    }
+
+    descSortData(data, key) {
+        return this.sortDataByKey(data, key, ((a, b) => {
+            if (Utils.parseFloatable(a) && Utils.parseFloatable(b)) {
+                a = Utils.parseIfFloat(a);
+                b = Utils.parseIfFloat(b);
+            }
+            if (a <= b) {
+                return 1;
+            } else if (a > b) {
+                return -1;
+            }
+        }).bind(this));
+    }
+
+    sortDataByKey(data, key, fn) {
+        const clone = [...data];
+
+        return clone.sort((a, b) =>
+            fn(a[key], b[key])
+        );
+    }
+
+    onStateChange(index) {
+        const sortings = this.state.sortings.map(((sorting, i) => {
+            if (i == index)
+                sorting = this.nextSortingState(sorting);
+
+            return sorting;
+        }).bind(this));
+
+        this.setState({
+            sortings
+        });
+    }
+
+    nextSortingState(state) {
+        let next;
+        switch (state) {
+            case "both":
+                next = "desc";
+                break;
+            case "desc":
+                next = "asc";
+                break;
+            case "asc":
+                next = "both";
+                break;
+        }
+        return next;
+    }
+
+    setHeaderWidths(widths){
+        if (!Array.isArray(widths)) throw new Error('widths is not an array');
+        this.setState({ 'widths' : widths });
+    }
+
+    render() {
+        const { data, columns, iconAsc, iconDesc, iconBoth } = this.props;
+        const { sortings, widths } = this.state;
+        _.each(columns, function (it, idx) { it.width = widths[idx]; });
```

Comment:
### NEVER MUTATE PROPS IN CHILD COMPONENTS

Read & understand [this article](https://codeburst.io/when-to-use-component-or-purecomponent-a60cfad01a81) in entirety, especially under header "NEVER MUTATE" which references this exact case.